bundles/af-core/src/main/java/com/adobe/cq/forms/core/components/internal/models/v2/form/FormContainerImpl.java

```diff
@@ -260,4 +260,17 @@ public Map<String, Object> getDorProperties() {
         }
         return customDorProperties;
     }
+
+    @Override
+    @JsonIgnore
+    public String getParentPagePath() {
+        if (resource != null) {
+            Resource parent = resource.getParent();
+            while (parent != null && !StringUtils.equals(parent.getResourceType(), "cq:Page")) {
```

Comment:
pls extract primary type of the resource 
either through valuemap or through node adaptation.